motionplan/plannerOptions.go

```diff
@@ -98,8 +98,8 @@ func newBasicPlannerOptions() *plannerOptions {
 	return opt
 }
 
-// plannerOptions are a set of options to be passed to a planner which will specify how to solve a motion planning problem.
-type plannerOptions struct {
+// PlannerOptions are a set of options to be passed to a planner which will specify how to solve a motion planning problem.
+type PlannerOptions struct {
```

Comment:
We should not be making this public; allowing users to hand-construct these structs is a disaster waiting to happen.

Instead, the public constructors for IK methods should take the same map[string]interface{} that motion.Solve does, and resolve those to the desired options the same way.

Reply:
Yeah I didn't like doing this either and did this as a temporary workaround so I could get feedback on the rest of the project. I like your idea better

Reply:
With the refactor, this comment should be addressed.  Going to resolve